src/js/trove/iolib.js

```diff
@@ -0,0 +1,47 @@
+({
+  requires: [ ],
+  provides: {
+    shorthands: { },
+    values: {
+      "prompt": ["arrow", ["String"], "String"]
```

Comment:
Have Haskell on my mind - in Haskell world this would be something like:

```hs
prompt :: IO String -> String
```

IO in Haskell took me a little while to wrap my head around. Fairly confident we don't want to introduce this here.